docs/scripts/api/buildComponentsDocumentation.ts

```diff
@@ -134,10 +134,10 @@ function extractSlots(options: {
       return;
     }
 
-    // Workaround to generate correct (camelCase) keys for slots in v6 `API Reference` documentation
-    // TODO v7: Remove camelCase once `Grid(Pro|Premium)SlotsComponent` type is refactored to have `camelCase` names
-    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedGridSlotsComponent`
-    const slotName = camelCase(name);
+    // Workaround to generate correct (camelCase) keys for slots in `API Reference` documentation
+    // TODO v7: Remove camelCase from pickers once the `components` prop is removed
+    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedSlotsComponent`
+    const slotName = project.name.includes('grid') ? name : camelCase(name);
 
     slots[slotName] = {
```

Comment:
@mui/explore we might need to do this once pickers also stop supporting `components/componentsProps`
```suggestion
    slots[name] = {
```

Reply:
It's ready #10700

Reply:
I merged this one, kindly consider merging the latest `next` and doing the diff mentioned in https://github.com/mui/mui-x/pull/10911#discussion_r1386277625 in the pickers' PR